### PR TITLE
More bug fixes

### PR DESCRIPTION
**Pull request by devongovett**
See commits.

### CODE REVIEW

File: packages/core/core/src/requests/AssetGraphRequest.js [hunk 1]
```diff
@@ -523,8 +523,10 @@ export class AssetGraphBuilder {
           if (assetGroups.length === 1) {
             let [assetGroupId] = assetGroups;
             let assetGroup = nullthrows(this.assetGraph.getNode(assetGroupId));
-            invariant(assetGroup.type === 'asset_group');
```
Comment (devongovett) [inline packages/core/core/src/requests/AssetGraphRequest.js:526]: It's possible for a dependency to point directly to an asset, without an asset group. This happens when an asset references another asset returned by the same transformer by uniqueKey. See `resolveAsset` in AssetGraph.js.

**Comment by parcel-benchmark** [comment]
## Benchmark Results
<details><summary>Kitchen Sink ✅</summary><p>

#### Timings

| Description | Time | Difference |
| --- | --- | --- |
| Cold | 1.55s | -78.00ms 🚀 |
| Cached | 397.00ms | -31.00ms 🚀 |

#### Cold Bundles

| Bundle | Size | Difference | Time | Difference |
| --- | --- | --- | --- | --- |
| dist/legacy/index.8a81a4b4.js | 2.00kb | +0.00b | 810.00ms | -43.00ms 🚀 |
| dist/modern/index.a053a4c6.js | 2.00kb | +0.00b | 810.00ms | -42.00ms 🚀 |
| dist/legacy/index.9effe70a.css | 77.00b | +0.00b | 810.00ms | -44.00ms 🚀 |
| dist/modern/index.ae61bfd0.css | 77.00b | +0.00b | 809.00ms | -45.00ms 🚀 |

#### Cached Bundles

| Bundle | Size | Difference | Time | Difference |
| --- | --- | --- | --- | --- |
| dist/legacy/parcel.7cdb0fad.webp | 102.94kb | +0.00b | 44.00ms | -11.00ms 🚀 |
| dist/modern/parcel.7cdb0fad.webp | 102.94kb | +0.00b | 71.00ms | +16.00ms ⚠️ |
| dist/legacy/index.8a81a4b4.js | 2.00kb | +0.00b | 71.00ms | -15.00ms 🚀 |
| dist/modern/index.a053a4c6.js | 2.00kb | +0.00b | 71.00ms | -15.00ms 🚀 |
| dist/legacy/index.html | 701.00b | +0.00b | 72.00ms | -14.00ms 🚀 |
| dist/modern/index.html | 701.00b | +0.00b | 71.00ms | -15.00ms 🚀 |
| dist/legacy/index.9effe70a.css | 77.00b | +0.00b | 72.00ms | -14.00ms 🚀 |
| dist/modern/index.ae61bfd0.css | 77.00b | +0.00b | 71.00ms | -15.00ms 🚀 |

</p></details>

<details><summary>React HackerNews ✅</summary><p>

#### Timings

| Description | Time | Difference |
| --- | --- | --- |
| Cold | 10.21s | +146.00ms |
| Cached | 647.00ms | +23.00ms |

#### Cold Bundles

*No bundle changes detected.*

#### Cached Bundles

*No bundle changes detected.*

</p></details>

<details><summary>AtlasKit Editor ✅</summary><p>

#### Timings

| Description | Time | Difference |
| --- | --- | --- |
| Cold | 1.14m | -1.83s |
| Cached | 2.69s | -243.00ms 🚀 |

#### Cold Bundles

| Bundle | Size | Difference | Time | Difference |
| --- | --- | --- | --- | --- |
| dist/Modal.275017ce.js | 45.31kb | +15.00b ⚠️ | 26.72s | -323.00ms |

#### Cached Bundles

| Bundle | Size | Difference | Time | Difference |
| --- | --- | --- | --- | --- |
| dist/index.65d11412.js | 2.60mb | +11.00b ⚠️ | 145.00ms | -25.00ms 🚀 |
| dist/pdf.worker.ebd8066b.js | 727.98kb | +0.00b | 201.00ms | -35.00ms 🚀 |
| dist/editorView.7771bd2f.js | 594.99kb | +0.00b | 216.00ms | -20.00ms 🚀 |
| dist/pdfRenderer.82551541.js | 411.56kb | +0.00b | 200.00ms | -36.00ms 🚀 |
| dist/popup.af65ac99.js | 209.85kb | +0.00b | 206.00ms | -30.00ms 🚀 |
| dist/EmojiPickerComponent.59865a5a.js | 147.28kb | +0.00b | 183.00ms | -10.00ms 🚀 |
| dist/Toolbar.fc3eb12c.js | 107.27kb | +0.00b | 217.00ms | -19.00ms 🚀 |
| dist/media-viewer.2d689646.js | 74.99kb | +0.00b | 200.00ms | -36.00ms 🚀 |
| dist/card.4de79179.js | 62.67kb | +0.00b | 189.00ms | -14.00ms 🚀 |
| dist/card.91b56cc7.js | 60.71kb | +0.00b | 210.00ms | -26.00ms 🚀 |
| dist/Modal.275017ce.js | 45.31kb | +15.00b ⚠️ | 151.00ms | -23.00ms 🚀 |
| dist/component.f8981c4c.js | 37.59kb | +0.00b | 150.00ms | -23.00ms 🚀 |
| dist/esm.fb31ee36.js | 33.24kb | +0.00b | 184.00ms | -18.00ms 🚀 |
| dist/component.e74ce88e.js | 24.95kb | +0.00b | 154.00ms | -73.00ms 🚀 |
| dist/DatePicker.3a33e0c3.js | 22.99kb | +0.00b | 179.00ms | -12.00ms 🚀 |
| dist/js.82b1c3de.js | 17.33kb | +0.00b | 152.00ms | -75.00ms 🚀 |
| dist/ui.4e71b166.js | 14.94kb | +0.00b | 187.00ms | -16.00ms 🚀 |
| dist/smartMediaEditor.cd3066ff.js | 13.25kb | +0.00b | 212.00ms | -24.00ms 🚀 |
| dist/dropzone.996e568b.js | 12.15kb | +0.00b | 208.00ms | -28.00ms 🚀 |
| dist/workerHasher.f2448c7e.js | 11.81kb | +0.00b | 185.00ms | -17.00ms 🚀 |
| dist/component.4e8fd03c.js | 6.82kb | +0.00b | 154.00ms | -72.00ms 🚀 |
| dist/card.4cdca260.js | 5.96kb | +0.00b | 190.00ms | -36.00ms 🚀 |
| dist/media-viewer.32e1b94c.js | 4.51kb | +0.00b | 201.00ms | -35.00ms 🚀 |
| dist/media-viewer.fbbe405b.js | 3.98kb | +0.00b | 201.00ms | -35.00ms 🚀 |
| dist/EmojiPickerComponent.8a87a9d0.js | 3.72kb | +0.00b | 186.00ms | -16.00ms 🚀 |
| dist/png-chunks-extract.1bcf0dfb.js | 3.58kb | +0.00b | 153.00ms | -74.00ms 🚀 |
| dist/index.46eb6237.css | 3.46kb | +0.00b | 223.00ms | -16.00ms 🚀 |
| dist/dropzone.0e17ef83.js | 3.29kb | +0.00b | 207.00ms | -29.00ms 🚀 |
| dist/Modal.7ade13df.js | 3.15kb | +0.00b | 150.00ms | -23.00ms 🚀 |
| dist/clipboard.c8afbeec.js | 2.93kb | +0.00b | 209.00ms | -27.00ms 🚀 |
| dist/16.01abc4bb.js | 2.36kb | +0.00b | 146.00ms | -25.00ms 🚀 |
| dist/ResourcedEmojiComponent.a1d3f4a0.js | 2.12kb | +0.00b | 186.00ms | -17.00ms 🚀 |
| dist/card.d1f2beda.js | 2.06kb | +0.00b | 202.00ms | -34.00ms 🚀 |
| dist/date.71db3fe8.js | 1.86kb | +0.00b | 157.00ms | -19.00ms 🚀 |
| dist/images.d62b15bd.js | 1.80kb | +0.00b | 160.00ms | -18.00ms 🚀 |
| dist/feedback.9958bdd4.js | 1.77kb | +0.00b | 172.00ms | -18.00ms 🚀 |
| dist/16.a4c09c54.js | 1.75kb | +0.00b | 219.00ms | -17.00ms 🚀 |
| dist/browser.330ba164.js | 1.69kb | +0.00b | 209.00ms | -27.00ms 🚀 |
| dist/workerHasher.c8ca1949.js | 1.63kb | +0.00b | 184.00ms | -18.00ms 🚀 |
| dist/workerHasher.c6d63a8c.js | 1.63kb | +0.00b | 211.00ms | -25.00ms 🚀 |
| dist/list-number.1094e49b.js | 1.59kb | +0.00b | 163.00ms | -19.00ms 🚀 |
| dist/status.01513cd7.js | 1.59kb | +0.00b | 167.00ms | -19.00ms 🚀 |
| dist/code.41dec3bc.js | 1.51kb | +0.00b | 156.00ms | -19.00ms 🚀 |
| dist/heading6.43c4a239.js | 1.51kb | +0.00b | 172.00ms | -17.00ms 🚀 |
| dist/heading3.86883cea.js | 1.49kb | +0.00b | 169.00ms | -19.00ms 🚀 |
| dist/link.cf6cf606.js | 1.43kb | +0.00b | 163.00ms | -19.00ms 🚀 |
| dist/16.fc05fc66.js | 1.40kb | +0.00b | 219.00ms | -16.00ms 🚀 |
| dist/heading5.eead537b.js | 1.38kb | +0.00b | 172.00ms | -17.00ms 🚀 |
| dist/emoji.209246d3.js | 1.36kb | +0.00b | 159.00ms | -19.00ms 🚀 |
| dist/16.c6b45ca4.js | 1.35kb | +0.00b | 155.00ms | -72.00ms 🚀 |
| dist/16.06271460.js | 1.33kb | +0.00b | 220.00ms | -16.00ms 🚀 |
| dist/16.04665c7c.js | 1.33kb | +0.00b | 147.00ms | -25.00ms 🚀 |
| dist/heading2.45d46a2f.js | 1.32kb | +0.00b | 169.00ms | -18.00ms 🚀 |
| dist/16.bc168422.js | 1.29kb | +0.00b | 220.00ms | -16.00ms 🚀 |
| dist/expand.44b4643e.js | 1.29kb | +0.00b | 172.00ms | -19.00ms 🚀 |
| dist/heading4.cc81126f.js | 1.27kb | +0.00b | 170.00ms | -18.00ms 🚀 |
| dist/16.c4b9ffab.js | 1.25kb | +0.00b | 219.00ms | -17.00ms 🚀 |
| dist/16.a9c08a24.js | 1.22kb | +0.00b | 147.00ms | -24.00ms 🚀 |
| dist/16.3be2d210.js | 1.21kb | +0.00b | 219.00ms | -17.00ms 🚀 |
| dist/16.c4ae24c0.js | 1.20kb | +0.00b | 155.00ms | -19.00ms 🚀 |
| dist/mention.bf94f359.js | 1.20kb | +0.00b | 164.00ms | -18.00ms 🚀 |
| dist/layout.2ab7fe1e.js | 1.18kb | +0.00b | 160.00ms | -19.00ms 🚀 |
| dist/Modal.41dab93a.js | 1.17kb | +0.00b | 152.00ms | -22.00ms 🚀 |
| dist/heading1.e46ebd87.js | 1.16kb | +0.00b | 168.00ms | -19.00ms 🚀 |
| dist/16.dfa6b620.js | 1.16kb | +0.00b | 147.00ms | -25.00ms 🚀 |
| dist/divider.4e444466.js | 1.16kb | +0.00b | 158.00ms | -19.00ms 🚀 |
| dist/quote.f1661da4.js | 1.15kb | +0.00b | 167.00ms | -18.00ms 🚀 |
| dist/16.d91bdfc5.js | 1.15kb | +0.00b | 218.00ms | -18.00ms 🚀 |
| dist/16.5247c925.js | 1.15kb | +0.00b | 154.00ms | -73.00ms 🚀 |
| dist/action.5318380d.js | 1.13kb | +0.00b | 155.00ms | -20.00ms 🚀 |
| dist/component.c218dc1f.js | 1.12kb | +0.00b | 146.00ms | -25.00ms 🚀 |
| dist/media-card-analytics-error-boundary.e1f2c03b.js | 1.12kb | +0.00b | 190.00ms | -16.00ms 🚀 |
| dist/decision.12c8af60.js | 1.12kb | +0.00b | 158.00ms | -19.00ms 🚀 |
| dist/panel-warning.bda6bdb5.js | 1.11kb | +0.00b | 166.00ms | -18.00ms 🚀 |
| dist/16.9ad179a5.js | 1.11kb | +0.00b | 146.00ms | +20.00ms ⚠️ |
| dist/list.f96161ec.js | 1.08kb | +0.00b | 163.00ms | -19.00ms 🚀 |
| dist/16.d102dfb8.js | 1.07kb | +0.00b | 220.00ms | -16.00ms 🚀 |
| dist/panel-error.02cac363.js | 1.01kb | +0.00b | 164.00ms | -18.00ms 🚀 |
| dist/panel.ba611692.js | 1.01kb | +0.00b | 166.00ms | -19.00ms 🚀 |
| dist/table.84de9360.js | 1022.00b | +0.00b | 168.00ms | -18.00ms 🚀 |
| dist/panel-success.c5920078.js | 978.00b | +0.00b | 165.00ms | -18.00ms 🚀 |
| dist/panel-note.671a06d0.js | 974.00b | +0.00b | 164.00ms | -19.00ms 🚀 |
| dist/media-viewer-analytics-error-boundary.e65539e8.js | 964.00b | +0.00b | 202.00ms | -34.00ms 🚀 |
| dist/media-picker-analytics-error-boundary.54a642f6.js | 964.00b | +0.00b | 208.00ms | -28.00ms 🚀 |
| dist/media-card-analytics-error-boundary.b22bb859.js | 960.00b | +0.00b | 189.00ms | -15.00ms 🚀 |
| dist/media-card-analytics-error-boundary.fb2a1470.js | 960.00b | +0.00b | 211.00ms | -25.00ms 🚀 |
| dist/simpleHasher.db850807.js | 641.00b | +0.00b | 185.00ms | -17.00ms 🚀 |
| dist/simpleHasher.660236a4.js | 641.00b | +0.00b | 212.00ms | -24.00ms 🚀 |
| dist/index.html | 119.00b | +0.00b | 100.00ms | -23.00ms 🚀 |

</p></details>

<details><summary>Three.js ✅</summary><p>

#### Timings

| Description | Time | Difference |
| --- | --- | --- |
| Cold | 7.28s | +86.00ms |
| Cached | 581.00ms | +20.00ms |

#### Cold Bundles

*No bundle changes detected.*

#### Cached Bundles

| Bundle | Size | Difference | Time | Difference |
| --- | --- | --- | --- | --- |
| dist/Three.js | 579.69kb | +0.00b | 71.00ms | +9.00ms ⚠️ |

</p></details>

**[Click here to view a detailed benchmark overview.](https://benchmark.parceljs.org/benchmark/300689259752325640)**

File: packages/core/core/src/requests/AssetGraphRequest.js (hunk 1 above)
Comment (mischnic) [reply packages/core/core/src/requests/AssetGraphRequest.js:526]: Did you hit this assertion? Should this actually be handled somehow and not just ignored?

Comment (devongovett) [reply packages/core/core/src/requests/AssetGraphRequest.js:526]: I did hit it due to our weird transformer that returns multiple assets and a dependency between them. I wasn't exactly sure how to handle it. What do you think?

Comment (mischnic) [reply packages/core/core/src/requests/AssetGraphRequest.js:526]: Yes, it should take these dependencies into account as well. I'll create a PR for that

**Review by mischnic — APPROVED** [review]

**Pull request merged**